src/apps/service/apps/de/admin.clj

```diff
@@ -80,26 +80,35 @@
   (when-not (nil? disabled)
     (persistence/disable-app disabled app-id)))
 
+(defn- update-app-extra-info
+  "Updates any extra information specific to execution platforms or similar."
+  [{app-id :id :keys [extra]}]
+  (when-let [htcondor (:htcondor extra)]
+    (when-let [extra-requirements (:extra_requirements htcondor)]
+      (persistence/set-htcondor-extra app-id extra-requirements))))
+
 (defn- update-app-details
   "Updates high-level details and labels in an App, including deleted and disabled flags in the
    database."
-  [{app-id :id :keys [references groups] :as app}]
+  [{app-id :id :keys [references groups extra] :as app}]
   (persistence/update-app app)
   (when-not (empty? references)
     (persistence/set-app-references app-id references))
   (when-not (empty? groups)
-    (update-app-labels (select-keys app [:id :groups]))))
+    (update-app-labels (select-keys app [:id :groups])))
+  (when-not (empty? extra)
+    (update-app-extra-info app)))
```

Comment:
It seems like we don't have a way to delete extra info.

If `set-htcondor-extra` is updated as suggested, so that an `insert` is not attempted if the value is empty, then this `when-not` check could be removed, and the `when-let` checks in `update-app-extra-info` could also be removed to allow a nil to be passed to `set-htcondor-extra` in order to remove the current value.

Reply:
I think it'll be sufficient to make the change to `set-htcondor-extra` -- `extra` won't be `empty?` unless there's nothing inside it, i.e.:

```
apps.core=> (empty? {:htcondor {:extra_requirements nil}})
false
apps.core=> (empty? {:htcondor {:extra_requirements ""}})
false
```

so with the change I'm about to push I think it'll be removed when an empty string is passed in, and that makes the endpoint behave more consistently with itself (if a key's that can be patched isn't included, don't touch it at all)

Reply:
cool, that makes sense too.